### ISSUE TITLE
Need search result based on matching keyword in products only

### ISSUE THREAD

**Issue by rahul-bhandary** [Smile-SA/elasticsuite#707]
Magento Version: 2.1.8 CE
ElasticSuite Version : 2.3.0
Environment: Developer mode
Third party modules: No

### Steps to reproduce

1. Enter any keyword in the search box
2. In the suggestion box, it shows search terms based on matching keyword and category name.
For eg., if I search 'men' then it shows search terms and products matching both having the keyword 'men' and also the one tagged under category 'men'.
3. On search result page also it shows search terms based on matching keyword and/or tagged under the category with the same name.

### Expected Result

I want search suggestions and search result based only on matching keyword in the product.
For eg., if I search 'men' then it should show search terms and products matching keyword 'men' only and skip the ones tagged under category 'men'.

### Actual result

In the search suggestions, it shows search terms based on matching keyword and/or tagged under the category with the same name.

### Additional comments

I know it makes sense to show category related products also but as per our requirement, we need only products with matching keyword in product name/meta name only.

Also, in search suggestions, I want to show only search terms and products. I want to remove categories from suggestions.

Please suggest if it is possible to achieve this through the module? If yes, what necessary changes I'll have to make?

**Comment by romainruaud** [comment]
Hello @rahul-bhandary 

This module is adding an attribute to each category, called "Use Category Name in product search" (use_name_in_product_search).

You can set this attribute to 0 for all categories, and set it's default value to be also 0 to prevent creating new categories with this setting activated.

This way you should achieve to disable this one for your whole website.

Best regards,

I close the issue, feel free to reopen it if needed

**Comment by rahul-bhandary** [comment]
Hi @romainruaud 

Thanks for the quick reply.

I set 'use_name_in_product_search' to 0 for all categories.
However, I found that for some search keywords it is showing some random products which are in no way related to the search term.

![search results for cover](https://user-images.githubusercontent.com/15210758/35331819-b33f68ea-012e-11e8-83b7-16772f4d62a7.png)


Also in search result page, it shows some random products with the message "No exact results found for: 'XYZ'. The displayed items are the closest matches." 
What if I want to show "No result found" when none of the words matches with the product name/meta keywords  i.e. search should consider only exact term match? How do I achieve this?
